builtin/joined_dataset.cc

```diff
@@ -59,6 +59,30 @@ struct JoinedDataset::Itl
         //ML::compact_vector<RowHash, 2> rowHashes;   ///< Row hash from input datasets
     };
 
+     struct JoinedRowStream : public RowStream {
+
+        JoinedRowStream(JoinedDataset::Itl* source) : source(source)
+        {
+            
+        }
+
+        virtual std::shared_ptr<RowStream> clone() const{
+            auto ptr = std::make_shared<JoinedRowStream>(source);
+            return ptr;
+        }
+
+        virtual void initAt(size_t start){
+            iter = source->rows.begin() + start;
+        }
+
+        virtual RowName next() {
```

Comment:
Followup story for the token? Or as we said we could skip it entirely and pass the boundselect (or a lambda of sort) to the dataset instead.